ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -569,6 +570,7 @@ abstract class EventStorageBackendTemplate(
       }(connection, loggingContext)
     } else {
       pruneWithLogging(queryDescription = "Archived retroactive divulgence events pruning") {
+        // Note: do not use `QueryStrategy.offsetIsSmallerOrEqual` because divulgence events have a nullable offset
```

Comment:
Unrelated to this PR: this query doesn't delete divulgence events with a null offset. Is this intended?

Reply:
Good question, it seems like we do not want to prune before-migration divulgences here. @tudor-da WDYT?

Reply:
Although this is a shady area. Pruning divulgences that have associated archivals after them is not fully sane as well, since a divulgee can still be blinded in command interpretation if they didn't witness the archival.

Reply:
Divulgence events with a NULL offset are those that were generated by the Flyway migration introducing the append-only schema. We can't reconstruct exactly when those divulgences happened, but they happened before the offset when the append-only migration took place (this offset is stored in the database).

I guess it's just something to keep in mind - divulgence events stored before the migration to the append-only schema are never deleted unless you enable `pruneAllDivulgedContracts`.